clients/fossid-webapp/src/main/kotlin/model/identification/markedAsIdentified/MarkedAsIdentifiedFile.kt

```diff
@@ -47,7 +47,7 @@ data class MarkedAsIdentifiedFile(
         val licenses = file.licenses?.let { licenses ->
             licenses.values.map {
                 License(
-                    identifier = it.file.licenseIdentifier!!,
+                    identifier = checkNotNull(it.file.licenseIdentifier),
```

Comment:
Mind giving a hint why `checkNotNull` is better than `!!``?

Reply:
Quite simply because Detekt does not complain anymore this way. Personally, I also find it more expressive than using the rather low-level `!!` "cast".

Reply:
So, the agreement would be to never use `!!`. For cases where `!!` is valid, we should instead use the semantically identical `!!` right ?

Reply:
> For cases where `!!` is valid, we should instead use the semantically identical `!!` right ?

I guess by the last `!!` you mean `checkNotNull`? Anyway, a "semantically required" `!!` should not always blindly be replaced with `checkNotNull`. You could also use `requireNotNull` depending on what you want to express. (Are you testing arguments or state? Also see https://bignerdranch.com/blog/write-better-code-using-kotlins-require-check-and-assert/.)

One of my problems with `!!` is that it does not really tell you *why* you can assume a value to be not-null; `checkNotNull` / `requireNotNull` are more expressive there, IMO.

Reply:
> I guess by the last !! you mean checkNotNull?

yes.

> One of my problems with !! is that it does not really tell you why you can assume a value to be not-null; checkNotNull / requireNotNull are more expressive there, IMO.

True, if a message is provided. So, given that I'm ok with ditching `!!` everywhere makes sense, so that at least we have it consistent even if there is no message provided.

Reply:
> True, if a message is provided.

I believe even without a message they are more expressive, as `checkNotNull` implies that something with the object state is wrong, while `requireNotNull` implies that an argument constraint was not met.